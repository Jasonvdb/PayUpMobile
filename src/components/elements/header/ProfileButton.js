import React, { Component } from "react";
import { StyleSheet, View, TouchableOpacity, Image } from "react-native";
import PropTypes from "prop-types";

class ProfileButton extends Component {
  onPressHandler = () => {
  	const { onPress } = this.props;

  	onPress();
  };

  render() {
  	return (
  		<TouchableOpacity onPress={this.onPressHandler}>
  			<View style={styles.root}>
  				<Image
  					style={styles.image}
  					source={{
  						uri: "https://api.adorable.io/avatars/285/jay"
  					}}
  				/>
  			</View>
  		</TouchableOpacity>
  	);
  }
}

ProfileButton.propTypes = {
	onPress: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
	root: {
		shadowColor: "black",
		shadowOffset: { width: 4, height: 4 },
		shadowOpacity: 0.25,
		shadowRadius: 4,
		borderWidth: 1,
		borderColor: "#F2F2F2",
		borderRadius: 16,
		borderStyle: "solid"
	},
	image: {
		borderRadius: 16,
		width: 30,
		height: 30
	}
});

export default ProfileButton;
