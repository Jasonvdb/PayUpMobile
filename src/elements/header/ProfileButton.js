import React from "react";
import { StyleSheet, View, Button, Image } from "react-native";
import PropTypes from "prop-types";

const ProfileButton = ({ onClick }) => {
	return (
		<View style={styles.root} onClick={onClick}>
			<Image
				style={styles.image}
				source={{
					uri: "https://api.adorable.io/avatars/285/jay"
				}}
			/>
		</View>
	);
};

ProfileButton.propTypes = {
	onClick: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
	root: {
		shadowColor: "black",
		shadowOffset: { width: 4, height: 4 },
		shadowOpacity: 0.25,
		shadowRadius: 4,
		borderWidth: 1,
		borderColor: "#F2F2F2",
		borderRadius: 16,
		borderStyle: "solid"
	},
	image: {
		borderRadius: 16,
		width: 30,
		height: 30
	}
});

export default ProfileButton;
