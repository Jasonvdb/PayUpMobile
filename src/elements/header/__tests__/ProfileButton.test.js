/**
 * @format
 */

import "react-native";
import React from "react";
import { shallow, render } from "enzyme";

import ProfileButton from "../ProfileButton";

describe("ProfileButton", () => {
	describe("Rendering", () => {
		it("should match to snapshot ", () => {
			const component = shallow(<ProfileButton onPress={() => {}}/>);
			expect(component).toMatchSnapshot();
		});
	});

	describe("Interaction", () => {
		describe("onPressHandler", () => {
			it("should call onPress", () => {
				// Arrange
				const mockOnPress = jest.fn(); // 1. mock function
				const component = shallow(
					<ProfileButton
						onPress={mockOnPress} // 2. passing in mock function as props
					/>,
				);
				const instance = component.instance(); // 3. getting an instance of component

				// Act
				instance.onPressHandler(); // 4. manually triggering onPressHandler()

				// Assert
				expect(mockOnPress).toHaveBeenCalled();
				expect(mockOnPress).toHaveBeenCalledTimes(1); // 5. checking return values
			});
		});
	});
});
