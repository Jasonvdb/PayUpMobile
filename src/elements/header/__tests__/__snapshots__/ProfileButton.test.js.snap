// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileButton Rendering should match to snapshot  1`] = `ShallowWrapper {}`;
